handling-forms/submitting-forms.md

```diff
@@ -0,0 +1,428 @@
+            emailData: newStudent.emailData,
+            isPresentData: false,
+        });
+
+        setStudentData(newStudentList);
+    };
+```
+<!-- prettier-ignore-end -->
+
+There are no rules about this `addStudentData` function, and Sofia could have written this logic using any approach or code style!
+
+Sofia chose to implement `addStudentData` this way:
+
+1. The function receives a new student object, `newStudent`.
+1. She duplicates the `studentData` array into `newStudentList`, which will help React detect a change.
+1. She generates a new ID number, `nextId`.
+1. She pushes a new object into `newStudentList`. The shape of this object matches the other objects in `studentData`.
+1. The `newStudentList` now contains an object with the `newStudent` data. She updates `studentData` in state with `setStudentData`.
+
+### !callout-info
+
+## The Implied Shape of `newStudent`
+
+Consider this implementation of `addStudentData` and how Sofia accessed the name and email of the `newStudent`.
+
+<br/>
+
+This implementation reaches into `newStudent` and grabs the values of `nameData` and `emailData` specifically... not `name`, or `newName`, or anything else! We should keep in mind that this function expects `newStudent` to have the keys exactly named `nameData` and `emailData`.
+
+<br/>
+
+If we want to change the name of the expected keys, we should refactor `App`'s definition of `addStudentData`.
+
+### !end-callout
+
+Now, Sofia needs to actually use her new function. She passes it into an instance of `NewStudentForm`, through a new prop named `addStudentCallback`.
+
+<!-- prettier-ignore-start -->
+```js
+    return (
+        <main>
+            <h1>Attendance</h1>
+            <StudentList
+                students={studentData}
+                onUpdateStudent={updateStudentData}
+            ></StudentList>
+            <NewStudentForm
+                addStudentCallback={addStudentData}
+            ></NewStudentForm>
+        </main>
+    );
+```
+<!-- prettier-ignore-end -->
+
+Sofia updates the PropTypes of `NewStudentForm` to now anticipate this prop:
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+
+const NewStudentForm = (props) => {
+// ...
+
+NewStudentForm.propTypes = {
+    addStudentCallback: PropTypes.func
+};
+```
+<!-- prettier-ignore-end -->
+
+## Handling Form Submissions 
+
+Sofia's next step is to get `NewStudentForm` to add student data when the form submits.
+
+She starts by creating a new function, `onFormSubmit`:
+
+<!-- prettier-ignore-start -->
+```js
+    const onFormSubmit = (event) => {
+        event.preventDefault();
+
+        props.addStudentCallback({
+            nameData: formFields.name,
+            emailData: formFields.email
+        });
+
+        setFormFields({
+            name: '',
+            email: '',
+        });
+    };
+```
+<!-- prettier-ignore-end -->
+
+Sofia's implementation of `onFormSubmit` looks like this:
```

Comment:
With the suggestion above, Sofia's name was recently mentioned, so switch this to Her.
Here we're explaining how the code works, not how it looks (that was given above)
```suggestion
Her implementation of `onFormSubmit` works like this:
```